parastell/magnet_coils.py

```diff
@@ -90,292 +144,223 @@ def logger(self):
+        tol = 2 * np.arctan2(self.max_cs_len, self.average_radial_distance)
 
         # Compute lower and upper bounds of toroidal extent within tolerance
-        min_rad = 2 * np.pi - tol
-        max_rad = self._toroidal_extent + tol
+        lower_bound = 2 * np.pi - tol
+        upper_bound = self._toroidal_extent + tol
 
-        for fil in self.filaments:
+        for coords, tangents in zip(
+            self.filament_coords, self.filament_tangents
+        ):
             # Compute filament center of mass
-            com = np.average(fil, axis=0)
+            com = np.average(coords, axis=0)
             # Compute toroidal angle of each point in filament
-            phi_pts = np.arctan2(fil[:, 1], fil[:, 0])
+            toroidal_angles = np.arctan2(coords[:, 1], coords[:, 0])
             # Ensure angles are positive
-            phi_pts = (phi_pts + 2 * np.pi) % (2 * np.pi)
+            toroidal_angles = (toroidal_angles + 2 * np.pi) % (2 * np.pi)
             # Compute bounds of toroidal extent of filament
-            min_phi = np.min(phi_pts)
-            max_phi = np.max(phi_pts)
+            min_tor_ang = np.min(toroidal_angles)
+            max_tor_ang = np.max(toroidal_angles)
 
             # Determine if filament toroidal extent overlaps with that of model
-            if (min_phi >= min_rad or min_phi <= max_rad) or (
-                max_phi >= min_rad or max_phi <= max_rad
+            if (min_tor_ang >= lower_bound or min_tor_ang <= upper_bound) or (
+                max_tor_ang >= lower_bound or max_tor_ang <= upper_bound
             ):
-                reduced_fils.append(fil)
+                filtered_coords.append(coords)
+                filtered_tangents.append(tangents)
                 com_list.append(com)
 
-        reduced_fils = np.array(reduced_fils)
+        filtered_coords = np.array(filtered_coords)
+        filtered_tangents = np.array(filtered_tangents)
         com_list = np.array(com_list)
 
         # Compute toroidal angles of filament centers of mass
-        phi_arr = np.arctan2(com_list[:, 1], com_list[:, 0])
-        phi_arr = (phi_arr + 2 * np.pi) % (2 * np.pi)
+        com_toroidal_angles = np.arctan2(com_list[:, 1], com_list[:, 0])
+        com_toroidal_angles = (com_toroidal_angles + 2 * np.pi) % (2 * np.pi)
 
-        # Sort filaments by toroidal angle
-        self.filtered_filaments = [
-            x for _, x in sorted(zip(phi_arr, reduced_fils))
-        ]
+        # Sort filaments by toroidal angle and overwrite respective arrays
+        self.filament_coords = np.array(
+            [x for _, x in sorted(zip(com_toroidal_angles, filtered_coords))]
+        )
+        self.filament_tangents = np.array(
+            [x for _, x in sorted(zip(com_toroidal_angles, filtered_tangents))]
+        )
+        self.filament_com = np.array(
+            [x for _, x in sorted(zip(com_toroidal_angles, com_list))]
+        )
```

Comment:
What if you calculated these in a setter function for `filament_coords`?

Reply:
Do you even need to store them, or could you calculate them inside the initializer for `MagnetCoil`?

Reply:
They could be calculated in `MagnetCoil`, but I would want all of the coordinate data available, to better resolve tangents and centers of mass. I think that combines well with your previous comment.

Reply:
I guess my point is that you don't use it until you build the coil.  It could even be temporary data that you generate when building the coil

Reply:
The centers of mass get used to sort the filaments, and unless you think it's worthwhile to recompute them at the coil stage rather than store them here, I've kept it as is. I've moved the computation of the tangents to the coil stage, however.